src/repr/src/adt/interval.rs

```diff
@@ -25,57 +25,65 @@ use crate::adt::numeric::DecimalLike;
 pub struct Interval {
     /// A possibly negative number of months for field types like `YEAR`
     pub months: i32,
-    /// A timespan represented in nanoseconds.
+    /// A possibly negative number of days.
     ///
-    /// Irrespective of values, `duration` will not be carried over into
+    /// Irrespective of values, `days` will not be carried over into `months`.
+    pub days: i32,
+    /// A timespan represented in microseconds.
+    ///
+    /// Irrespective of values, `micros` will not be carried over into `days` or
     /// `months`.
-    pub duration: i128,
+    pub micros: i64,
```

Comment:
Comment/Question: One consequence of this change is that Interval's precision decreases from nanosecond to microsecond Is this something we care about? The only potentially negative thing I've noticed so far is that we use Interval's to print out some of the timing output of `EXPLAIN` queries. This results in the last couple of decimal places being truncated.

Reply:
PG doesn't return nanoseconds, so I believe this is totally fine. The first implementation of interval in Materialize was _not_ focused on PG interoperability, so had a number of differences; this is likely something from that first iteration that I didn't think to make consistent with PG during my refactor of it.

One thing this did get me to realize, though, is that sub microsecond values do get rounded to microseconds:

```sql
SELECT INTERVAL '01:00:01.00000009';
 interval 
----------
 01:00:01
```

I'm still reviewing, but this looks like something we definitely want to ensure we test/catch.

Reply:
Good point, I've added the following tests for this and it seems to be consistent with Postgres as well.
```
query T
SELECT INTERVAL '01:00:01.00000009';
----
01:00:01

query T
SELECT INTERVAL '01:00:01.0000009';
----
01:00:01.000001

query T
SELECT INTERVAL '01:00:01.0000004';
----
01:00:01
```